ssg/templates.py

```diff
@@ -316,6 +317,40 @@ def build_rule(self, rule_id, rule_title, template, langs_to_generate, identifie
                 raise e(
                     "Error building templated {0} content for rule {1}".format(lang, rule_id))
 
+    def build_platform(self, platform):
+        for symbol in platform.test.get_symbols():
+            cpe_id = symbol.as_id()
+            cpe = self.product_cpes.get_cpe(cpe_id)
+
+            if 'name' not in cpe.template:
+                continue
+            template_name = cpe.template['name']
+            arg = symbol.arg
+            if arg in cpe.args:
+                rendered_vars = cpe.args[arg]
+            elif arg is None:
+                rendered_vars = {}
+            else:
+                raise ValueError("Platform {0} does not allow arg '{1}' ".format(symbol.name, arg))
+            rendered_vars.update(symbol.as_dict())
+            template_vars = self.process_product_vars(rendered_vars)
+
+            # Add the rule_id ID which will be reused in OVAL templates as OVAL
+            # definition ID so that the build system matches the generated
+            # check with the rule.
+            template_vars["_rule_id"] = cpe_id
+            # checks and remediations are processed with a custom YAML dict
+            local_env_yaml = self.env_yaml.copy()
+            local_env_yaml["rule_id"] = cpe_id
+            local_env_yaml["rule_title"] = cpe.title
+            local_env_yaml["products"] = self.env_yaml["product"]
```

Comment:
This is basically a copy-paste from `get_lang_for_rule` - along with misleading comments. The template interface should be altered, so it blends with the platform use case.